input/script_utils.go

```diff
@@ -923,14 +925,28 @@ func CommitSpendNoDelay(signer Signer, signDesc *SignDescriptor,
 	}
 
 	// Finally, we'll manually craft the witness. The witness here is the
-	// exact same as a regular p2wkh witness, but we'll need to ensure that
-	// we use the tweaked public key as the last item in the witness stack
-	// which was originally used to created the pkScript we're spending.
+	// exact same as a regular p2wkh witness, depending on the value of the
+	// tweakless bool.
 	witness := make([][]byte, 2)
 	witness[0] = append(sweepSig, byte(signDesc.HashType))
-	witness[1] = TweakPubKeyWithTweak(
-		signDesc.KeyDesc.PubKey, signDesc.SingleTweak,
-	).SerializeCompressed()
+
+	// TODO(roasbeef): optionally can make this implicit, by checking if
+	// the SingleTweak is populated or not.
+
+	switch tweakless {
+	// If we're tweaking the key, then we use the tweaked public key as the
+	// last item in the witness stack which was originally used to created
+	// the pkScript we're spending.
+	case false:
+		witness[1] = TweakPubKeyWithTweak(
```

Comment:
I wonder is this PR would be simpler if we just changed `TweakPubKeyWithTweak` to be a no-op when given a empty `SingleTweak`? 

That way most of the structs and logic would stay the same, we would only set a zero/empty tweak for these channels.

Reply:
I considered that, but instead opted for a more explicit API design to force the caller to recognize which they of witness they're requesting.